### DISCUSSION TITLE
Android NativeAOT

### DISCUSSION THREAD

**Discussion by gabriel-kozma** [AvaloniaUI/Avalonia#17015] [Q&A]
Hey there, I'm considering using Avalonia for migrating a project i have that's written in Xamarin.Forms and wanted to ask, how can I achieve this performance: https://www.reddit.com/r/dotnet/comments/13lvih2/nativeaot_ndk_vs_xamarinandroid_performance/

**Comment by thevortexcloud** [comment]
It has not been released yet. It was recently announced that it will be a paid feature.

https://github.com/AvaloniaUI/Avalonia/discussions/16997

From memory there is also some work happening in the .NET runtime itself for android AOT. But I don't know when that will ship.

**Comment by gabriel-kozma** [comment]
thank you for the answer @thevortexcloud I've filled the form with the research, but from what I gathered, this will be available only on a latter date, correct?

If/when you guys launch the subscription model, could I get a beta of this? It's the most important feature for me and the reason for subscribing

**Reply by thevortexcloud** [reply]
The android functionality is listed as being a phase 3 project. Which means it probably wont be out for awhile. That said they do say:

> Early Access: We're aiming to have some tools available for early access by the end of this year.

So I guess sign up for the beta and _gently_ lobby for it to be a higher priority to get it bumped to phase 2? I suspect you could pay for them to directly work on the feature now if you really need it done faster. But you would need to discuss that with the core team.

**Comment by gabriel-kozma** [comment]
@MikeCodesDotNET if I signup for `Avalonia Accelerate` will I be able to beta test the Android AOT part? I wanted to ask this on the feedback request you've posted, but it is locked now
